File: SpringBoot/AnotherProject/src/main/java/com/richard/estudos/anotherproject/services/UsersService.java
package com.richard.estudos.anotherproject.services;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.richard.estudos.anotherproject.daos.UsersDao;
import com.richard.estudos.anotherproject.models.User;

@Service
public class UsersService {
	@Autowired
	private UsersDao usersDao;
	
	@Autowired
	private CategoriesService categoriesService;
	
	public Iterable<User> findAll() {
		return usersDao.findAll();
	}
	
	public User save(User user) {
		categoriesService.findById(user.getCategory().getId());
		return usersDao.save(user);
	}
}

File: Kotlin/dynkot/src/main/kotlin/com/richard/dynkot/services/UsersService.kt
package com.richard.dynkot.services

import com.richard.dynkot.entities.User
import com.richard.dynkot.errors.NotFoundException
import com.richard.dynkot.repositories.UsersRepository
import org.springframework.stereotype.Service

@Service
class UsersService(private val usersRepository : UsersRepository) {

    fun findByEmail(email : String) : User {
        val user = usersRepository.findById(email)
        if( user.isPresent ) {
            throw NotFoundException(email)
        }
        return user.get()
    }

    fun save(user: User) : User {
        return usersRepository.save(user)
    }
}
